test/js/node/test/parallel/test-async-hooks-vm-gc.js

```diff
@@ -0,0 +1,15 @@
+// Flags: --expose-gc
+'use strict';
+
+require('../common');
+const asyncHooks = require('async_hooks');
+const vm = require('vm');
+
+// This is a regression test for https://github.com/nodejs/node/issues/39019
+//
+// It should not segfault.
+
+const hook = asyncHooks.createHook({ init() {} }).enable();
```

Comment:
lol i think i didn't include this test before because we don't actually implement createHook